### PR TITLE
Poseidon examples 

### PR DESCRIPTION
**Pull request by ImmanuelSegol**

### CODE REVIEW

**Comment by ImmanuelSegol** [comment]
@svpolonsky im adding Poseidon rust and c++ examples

File: examples/c++/Poseidon-hash/README.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+```c++
+#include "appUtils/poseidon/poseidon.cu"
+...
+poseidon_hash<scalar_t, arity+1>(input, output, n, constants, config);
+```
+
+**Parameters:**
+
+- **`scalar_t`:** a scalar field of the selected curve. Currently only `BLS12-381`.
```
[the remark this answers is not in the document]
Comment (svpolonsky) [reply examples/c++/Poseidon-hash/README.md:21]: fixed

File: examples/c++/Poseidon-hash/example.cu [hunk 2]
```diff
@@ -0,0 +1,153 @@
+#include <chrono>
+#include <fstream>
+#include <iostream>
+
+// select the curve (only 2 available so far)
```
[the remark this answers is not in the document]
Comment (svpolonsky) [reply examples/c++/Poseidon-hash/example.cu:5]: fixed

File: examples/c++/Poseidon-hash/example.cu [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    d = d + scalar_t::one();
+  }
+  std::cout << "Hashing blocks into tree leaves..." << std::endl;
+  PoseidonConstants<scalar_t> constants;
+  init_optimized_poseidon_constants<scalar_t>(data_arity, ctx, &constants);
+  PoseidonConfig config = default_poseidon_config<scalar_t>(data_arity+1); 
+  poseidon_hash<curve_config::scalar_t, data_arity+1>(data, &tree[tree_index(leaf_level, 0)], tree_width, constants, config);
+
+  std::cout << "3. Building Merkle tree" << std::endl;
+  // Poseidon<BLS12_381::scalar_t> tree_poseidon(tree_arity, stream);
```
[the remark this answers is not in the document]
Comment (svpolonsky) [reply examples/c++/Poseidon-hash/example.cu:124]: done

**Comment by jeremyfelder** [comment]
Closing in favor of #392 and #375

**Pull request closed without merging**